File: sumar.rb
def sumar(nums, index)
  total =0
  if index == nums.length - 1
    total += nums[index]
  else
    total = nums[index] +  sumar(nums, index + 1)
  end
  total
end
puts "TOTAL #{sumar([1,2,3], 0)}"

File: sumar.exs
defmodule SumarArray do
 def sum(list), do: _sum(list, 0)
 defp _sum([], total), do: total
 defp _sum([head | tail], total), do: _sum(tail, head+total)
end
IO.puts "TOTAL #{SumarArray.sum([1,2,3])}"
